drivers/hmis_csv_importer/app/models/hmis_csv_importer/importer/import_change_calculator.rb

```diff
@@ -0,0 +1,92 @@
+module HmisCsvImporter::Importer
+  class ImportChangeCalculator
+    def self.apply(...) = new.apply(...)
+
+    # Calculate changes for a single file/class combination
+    # @param klass [Class] The class representing the data model
+    # @param existing_scope [ActiveRecord::Relation] The scope for existing data
+    # @param incoming_scope [ActiveRecord::Relation] The scope for incoming data
+    # @return [Hash] Hash containing counts of records to add and remove
+    def apply(klass:, existing_scope:, incoming_scope:)
+      results = {}
+      return results unless klass.respond_to?(:hud_key)
+
+      @connection = klass.connection
+      temp_table_name = nil
+      begin
+        # Create temp table for incoming data
+        temp_table_name = create_temp_table(klass, incoming_scope)
+
+        # Calculate additions and removals
+        results['added'] = count_additions(klass, existing_scope, temp_table_name)
+        results['removed'] = count_removals(klass, existing_scope, temp_table_name)
+      ensure
+        # Clean up temp table even if an error occurs
+        drop_temp_table(temp_table_name)
+      end
+      results
+    end
+
+    private
+
+    # Create a temporary table with incoming data
+    def create_temp_table(klass, incoming_scope)
```

Comment:
I wonder if it would make sense to keep these around for more of the process than just the counting.  I was also hoping we could pull in the primary key so we could later join back to the source and destination tables in a more performant way (though maybe `where data_source_id = 1 and ProjectID = '1bc'` is performant enough.)  
If we do the calculations here, and store a table of ids for each process, then later when we go to make the updates/additions/deletions it could just join the tmp table to the appropriate other tables instead of having to recalculate the entire scope.